swingset/src/main/java/com/nqadmin/swingset/SSTextField.java

```diff
@@ -287,21 +296,44 @@ public SSTextField(final int _mask, final int _numberOfDecimalPlaces, final int
 	 *                         bound
 	 */
 	public SSTextField(final RowSet _rowSet, final String _boundColumnName) {
-		this();
-		bind(_rowSet, _boundColumnName);
+		this(null, _rowSet, _boundColumnName,
+			 DEFAULT_MASK, DEFAULT_NUMBER_OF_DECIMAL_PLACES, ANY_ALIGNMENT);
 	}
 
-	/**
-	 * Constructs a new text field with the specified text and mask.
-	 *
-	 * @param _text the text to be displayed.
-	 * @param _mask the mask required for this textfield.
-	 */
-	public SSTextField(final String _text, final int _mask) {
+	/** All the contructors feed through here */
+	private SSTextField(String _text, final RowSet _rowSet, final String _boundColumnName,
+	final int _mask, final int _numberOfDecimalPlaces, final int _align) {
 		super(_text);
```

Comment:
I wonder if we need both ssCommon and constructingSSCommon.

Rather than this:
```
	private SSCommon constructingSSCommon() {
		if (constructingSSCommon == null) {
			constructingSSCommon = new SSCommon(this);
		}
		return constructingSSCommon;
	}
```

Could we use this:
```
	private SSCommon constructingSSCommon() {
                SSCommon result = ssCommon;

		if (ssCommon == null) {
			result = new SSCommon(this);
		}
		return result;
	}
```

Reply:
`constructingSSCommon()` is used from `createDefaultModel()` which needs to use the same SSCommon that ends up in the variable ssCommon. 

The real issue is that `JTextField` calls `createDefaultModel()` in its constructor; and the `SSTextField` constructor does `super(_text)` so the document model, which depends on sscommon, has to be in place before `SSTextField` is constructed. 

An alternative approach would be to let `JTextField` use it's own document model, then in `SSTextField`'s constructor create the real model, based on sscommon, copy the text from the `JTextField`'s model into the real model, and set the new model into `SSTextField`. That would be more work (creating two models) and since I tend to overoptimize...

I added a comment, which _may_ clarify things at least a little bit.